backend/src/router.ts

```diff
@@ -80,41 +91,67 @@ router.post("/defence/configure", (req, res) => {
   // id of the defence
   const defenceId: DEFENCE_TYPES = req.body?.defenceId;
   const config: DefenceConfig[] = req.body?.config;
-  if (defenceId && config) {
+  const phase: PHASE_NAMES = req.body?.phase;
+  if (defenceId && config && phase >= 0) {
     // configure the defence
-    req.session.defences = configureDefence(
+    req.session.phaseState[phase].defences = configureDefence(
       defenceId,
-      req.session.defences,
+      req.session.phaseState[phase].defences,
       config
     );
     res.send("Defence configured");
   } else {
     res.statusCode = 400;
-    res.send("Missing defenceId or config");
+    res.send("Missing defenceId or config or phase");
   }
 });
 
 // reset the active defences
 router.post("/defence/reset", (req, res) => {
-  req.session.defences = getInitialDefences();
-  console.debug("Defences reset");
-  res.send("Defences reset");
+  const phase: PHASE_NAMES = req.body?.phase;
+  if (phase >= 0) {
```

Comment:
Could we just have `if (phase) {` here? And other places where we check for `if (phase >= 0) {`

Reply:
i did have this but it returns false when phase = 0 and gives a 400 

![image](https://github.com/gsproston-scottlogic/prompt-injection/assets/118981273/0d3659bc-a87f-4379-9463-e2af43c9d2fe)

Reply:
ahh I see. In other parts of the code we've got 
```
  const phase: number | undefined = req.query?.phase as number | undefined;
  if (phase) {
  ```
  Would that work?

Reply:
yeah it seem to be fine when inside the get request url 

![image](https://github.com/gsproston-scottlogic/prompt-injection/assets/118981273/db7a466d-1a26-408c-a1a8-28e77c0fb71d)